capsules/src/symmetric_encryption.rs

```diff
@@ -137,9 +137,45 @@ impl<'a, E: SymmetricEncryptionDriver + 'a> Crypto<'a, E> {
             state: Cell::new(CryptoState::IDLE),
         }
     }
+
+    fn aes128_crypt_ctr(&self, state: CryptoState) -> ReturnCode {
+        if self.key_configured.get() && !self.busy.get() && self.state.get() == CryptoState::IDLE {
+            for cntr in self.apps.iter() {
+                cntr.enter(|app, _| {
+                    self.busy.set(true);
+                    self.state.set(state);
+                    app.data_buf.as_ref().map(|slice| {
+                        let len1 = slice.len();
+                        // Copy data from app slice to kernel buffer.
+                        self.kernel_data.take().map(|buf| {
+                            for (out, inp) in buf.iter_mut().zip(slice.as_ref()[0..len1].iter()) {
+                                *out = *inp;
+                            }
+                            app.ctr_buf.as_ref().map(|slice2| {
+                                let len2 = slice2.len();
+                                // Copy counter value to kernel buffer.
+                                self.kernel_ctr.take().map(move |ctr| {
+                                    for (out, inp) in ctr.iter_mut()
+                                        .zip(slice2.as_ref()[0..len2].iter()) {
+                                        *out = *inp;
+                                    }
+                                    self.crypto.aes128_crypt_ctr(buf, ctr, len1);
```

Comment:
Looks like that I have been sloppy :(
I think we need to add an assertion here i.e., the initial counter should be 16 bytes for aes128, 24 bytes for aes192 and 32 byte for aes256.
Maybe better for now just to check if len2 > 16 and iterate from 0..16 since we only support aes128 right?